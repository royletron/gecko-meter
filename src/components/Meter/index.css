$bg-color: #2A2A2A;
$fg-color: #DFDFDF;

svg.this {
  background-color: $bg-color;
  > text {
    fill: #DFDFDF;
    font-family: sans-serif;
  }
}
