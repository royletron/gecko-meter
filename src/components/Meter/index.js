import React from 'react';
import classNames from 'classnames';

import styles from './index.css';

export default class Meter extends React.Component {
  render () {
    return (
      <div>I want to grow up to be a meter</div>
    );
  }
}
