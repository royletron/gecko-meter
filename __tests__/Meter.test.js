'use strict';

import React from 'react';
import Link from '../Link';
import Meter from '../src/components/Meter';
import renderer from 'react-test-renderer';

it('should display the correct values', () => {
  const component = renderer.create(
    <Meter min={10} max={100} value={40} />
  )
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

it('should display with prefix', () => {
  const component = renderer.create(
    <Meter pre='Â£' min={10} max={100} value={40} />
  )
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
