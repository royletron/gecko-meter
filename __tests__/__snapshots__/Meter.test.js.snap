exports[`test should display the correct values 1`] = `
<svg
  className=""
  height="180"
  width="300">
  <text
    className=""
    x="150"
    y="10">
    
    40
  </text>
  <path
    className=""
    d="M0,90 A90,90,0 1 1 180,90"
    transform="translate(60,55)" />
  <text
    x="60"
    y="155">
    
    10
  </text>
  <text
    x="240"
    y="155">
    
    100
  </text>
  <g
    className=""
    transform="translate(150, 148)">
    <line
      className={undefined}
      transform="rotate(-30 0 0)"
      x1="0"
      x2="0"
      y1="0"
      y2="-110" />
    <line
      transform="rotate(-30 0 0)"
      x1="0"
      x2="0"
      y1="0"
      y2="-95" />
  </g>
</svg>
`;

exports[`test should display with prefix 1`] = `
<svg
  className=""
  height="180"
  width="300">
  <text
    className=""
    x="150"
    y="10">
    £
    40
  </text>
  <path
    className=""
    d="M0,90 A90,90,0 1 1 180,90"
    transform="translate(60,55)" />
  <text
    x="60"
    y="155">
    £
    10
  </text>
  <text
    x="240"
    y="155">
    £
    100
  </text>
  <g
    className=""
    transform="translate(150, 148)">
    <line
      className={undefined}
      transform="rotate(-30 0 0)"
      x1="0"
      x2="0"
      y1="0"
      y2="-110" />
    <line
      transform="rotate(-30 0 0)"
      x1="0"
      x2="0"
      y1="0"
      y2="-95" />
  </g>
</svg>
`;
